samples/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient/Dockerfile

```diff
@@ -1,22 +1,31 @@
-RUN dotnet restore "samples/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient.csproj"
+WORKDIR /
+COPY "NHSD.BuyingCatalogue.Identity-ExludeDatabase.sln" "NHSD.BuyingCatalogue.Identity-ExludeDatabase.sln"
+COPY "samples/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient.csproj" "samples/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient.csproj"
+COPY "samples/NHSD.BuyingCatalogue.Identity.Api.SampleResource/NHSD.BuyingCatalogue.Identity.Api.SampleResource.csproj" "samples/NHSD.BuyingCatalogue.Identity.Api.SampleResource/NHSD.BuyingCatalogue.Identity.Api.SampleResource.csproj"
+COPY "samples/NHSD.BuyingCatalogue.Identity.Api.SampleClient/NHSD.BuyingCatalogue.Identity.Api.SampleClient.csproj" "samples/NHSD.BuyingCatalogue.Identity.Api.SampleClient/NHSD.BuyingCatalogue.Identity.Api.SampleClient.csproj"
+COPY "src/NHSD.BuyingCatalogue.Organisations.Api/NHSD.BuyingCatalogue.Organisations.Api.csproj" "src/NHSD.BuyingCatalogue.Organisations.Api/NHSD.BuyingCatalogue.Organisations.Api.csproj"
+COPY "tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/NHSD.BuyingCatalogue.Identity.Api.UnitTests.csproj" "tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/NHSD.BuyingCatalogue.Identity.Api.UnitTests.csproj"
+COPY "src/NHSD.BuyingCatalogue.Identity.Common/NHSD.BuyingCatalogue.Identity.Common.csproj" "src/NHSD.BuyingCatalogue.Identity.Common/NHSD.BuyingCatalogue.Identity.Common.csproj"
+COPY "src/NHSD.BuyingCatalogue.Identity.Api/NHSD.BuyingCatalogue.Identity.Api.csproj" "src/NHSD.BuyingCatalogue.Identity.Api/NHSD.BuyingCatalogue.Identity.Api.csproj"
+COPY "tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.csproj" "tests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests/NHSD.BuyingCatalogue.Identity.Api.IntegrationTests.csproj"
+COPY "tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests.csproj" "tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests.csproj"
+COPY "tests/NHSD.BuyingCatalogue.Identity.Api.Testing.Data/NHSD.BuyingCatalogue.Identity.Api.Testing.Data.csproj" "tests/NHSD.BuyingCatalogue.Identity.Api.Testing.Data/NHSD.BuyingCatalogue.Identity.Api.Testing.Data.csproj"
+RUN dotnet restore "NHSD.BuyingCatalogue.Identity-ExludeDatabase.sln"
+
 COPY . .
-WORKDIR "/src/samples/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient"
-RUN dotnet build "NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient.csproj" -c Release -o /app/build
+WORKDIR "/samples/NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient"
+RUN dotnet publish --no-restore -c Release -o /app
 
-FROM build AS publish
-RUN dotnet publish "NHSD.BuyingCatalogue.Identity.Api.SampleMvcClient.csproj" -c Release -o /app/publish
+FROM build as publish
```

Comment:
This is an empty stage I believe. I'd either
a) get rid of it and rename the copy statement in the final stage
b) move the dotnet publish step here